pkg/pool-weighted/contracts/WeightedPool.sol

```diff
@@ -343,26 +343,20 @@ contract WeightedPool is BaseWeightedPool, WeightedPoolProtocolFees {
         return supply.add(protocolFeesToBeMinted);
     }
 
-    /**
-     * @notice Returns the appreciation of one BPT relative to the underlying tokens.
-     * @dev This is equivalent to `BaseWeightedPool.getRate()`, with a correction factor to the total supply.
-     * We add on the to-be-minted protocol fees to the total supply to dilute the value of the remaining BPT.
-     * This prevents the Pool's rate being affected by the collection of protocol fees.
-     */
-    function getRate() public view override returns (uint256) {
-        uint256 supply = totalSupply();
-        uint256 invariant = getInvariant();
-        (uint256 protocolFeesToBeMinted, ) = _getPreJoinExitProtocolFees(invariant, _getNormalizedWeights(), supply);
-        return Math.mul(invariant, _getTotalTokens()).divDown(supply.add(protocolFeesToBeMinted));
-    }
-
     function _onDisableRecoveryMode() internal override {
         // Update the postJoinExitInvariant to the value of the currentInvariant, zeroing out any protocol swap fees.
         _updatePostJoinExit(getInvariant());
 
-        // Update the athRateProduct to the value of the current rateProduct, zeroing out any protocol yield fees.
+        // If the Pool has any protocol yield fees accrued then we update the athRateProduct to zero these out.
+        // If the current rate product is less than the athRateProduct then we do not perform this update, this prevents
+        // the Pool from paying protocol fees on the same yield twice.
```

Comment:
Could also expand a little on the last line. Not checking could result in lowering the athRateProduct, which is how it can result in paying fees twice on the difference between the current rate and previous ATH.